### PR TITLE
feat: add service account support

### PR DESCRIPTION
**Pull request by sighphyre**
Adds a resource layer for service accounts, this does not alter or bind tokens in any way (coming soon to a PR near you)

### CODE REVIEW

File: internal/provider/service_account_resource.go [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+		Description: "Manages a service account.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.Int64Attribute{
+				Description: "The ID of the service account.",
+				Computed:    true,
+			},
+			"username": schema.StringAttribute{
+				Description: "The username for the service account.",
+				Optional:    true,
+			},
```
Comment (gastonfournier) [inline internal/provider/service_account_resource.go:64]: I'll raise another ticket into our project to improve OpenAPI. This field is not Optional according to our UI:
Screenshot from 2024-08-28 10-21-09


But in our open api spec it looks like both name and username are non mandatory:
image

File: internal/provider/service_account_resource.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+	resp.Schema = schema.Schema{
+		Description: "Manages a service account.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.Int64Attribute{
+				Description: "The ID of the service account.",
+				Computed:    true,
+			},
+			"username": schema.StringAttribute{
+				Description: "The username for the service account.",
+				Optional:    true,
```
Comment (gastonfournier) [inline internal/provider/service_account_resource.go:63]: Maybe we can make it required for terraform:
```suggestion
				Required:    true,
```

File: internal/provider/service_account_resource.go [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+
+	roleId, diags := plan.RootRole.ToInt64Value(ctx)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	plan.RootRole.ToInt64Value(ctx)
```
Comment (gastonfournier) [inline internal/provider/service_account_resource.go:98]: Is this line needed?

**Review by gastonfournier — APPROVED** [review]
LGTM!

File: internal/provider/service_account_resource.go (hunk 3 above)
Comment (sighphyre) [reply internal/provider/service_account_resource.go]: Nope, nice catch, removed

File: internal/provider/service_account_resource.go (hunk 2 above)
Comment (sighphyre) [reply internal/provider/service_account_resource.go]: Cheers, suggestion taken.

Turns out it's required on the API level too. So you get blocked by the validation if it's not set

Name on the other hand is marked as required on the UI. Not required on the API and not required on the OpenAPI spec

:upside_down_face:

**Pull request merged**